app.py

```diff
@@ -1,50 +1,150 @@
+    empty.
+    Parameters:
+    - session: Flask session object
+    '''
+   
+    if 'jobs' not in session:
+        logger.info('Empty session')
+        return True
+
+    def clear_session():
+        for idx, job_dict in enumerate(session['jobs']):
+            job = q.fetch_job(job_dict['id'])
+            video_source = 'static/videos/{}'.format(job_dict['tracks_filename'])
+            
+            dead_job_or_timeout = job is None or job.get_status() == 'failed'
+            video_does_not_exist = not os.path.exists(video_source)
+            
+            if dead_job_or_timeout and video_does_not_exist:
+                session['jobs'].pop(idx)
+                # Have to set modified to True after every in-place operation
+                session.modified = True 
+                logger.info('Popped job from session: id={}'.format(job_dict['id']))
+
+    clear_session()
+    
+    if len(session['jobs']) == 0:
+        logger.info('Empty session')
+        return True
+    
+    return False
 
-jobs = []
+@app.template_filter('refresh_job')
+def refresh_job(job_dict):
+    '''
+    This function updates the meta dictionary of a given job and returns the
+    job's filename
+    Parameters:
+    - job_dict: Dictionary containing a job id and job filename
+    '''
+    logger.info('Refresh job, id={}'.format(job_dict['id']))
+    job = q.fetch_job(job_dict['id'])
+    filename = job_dict['filename']
+    try:
+        job.refresh()
+        logger.info('Job refreshed successfully')
+    except Exception as e: 
+        logger.info('Job did not refresh properly, exception {}'.format(e))
+    return filename
 
+@app.template_filter('display_job')
+def display_job(job_dict):
+    '''
+    This function checks whether or not a video with the job's track filename 
+    exists. If it exists, then it returns HTML markup for the video source.
+    Else, it returns the progress information.
+    Parameters:
+    - job: Dictionary containing a job id and tracks filename
+    '''
+    logger.info('Displaying job, id={}'.format(job_dict['id']))
+    video_source = 'static/videos/{}'.format(job_dict['tracks_filename'])
+    if os.path.exists(video_source):
+        return Markup("""
+        <video width="320" height="240" controls>
+          <source src="/{}" type="video/mp4">
+        </video>
+        """.format(video_source))
+    job = q.fetch_job(job_dict['id'])
+    if job and job.meta.get('status'): 
+        return job.meta.get('status')
+    return 'Beginning process...'
+
+@app.route('/<path:path>')
+def send_static(path):
+    return send_from_directory('', path)
 
 @app.route('/upload', methods=['POST'])
 def upload():
-
     logger.info(f'upload() request.files: {request.files}')
-
     video_file = request.files.get('file')
     fname_video = video_file.filename
     video_stream = video_file.read()
     with open(fname_video, 'wb') as f:
-      f.write(video_stream)
-
+        f.write(video_stream)
+    
```

Comment:
abrichr 9 days ago Member
Please remove spurious whitespace.

See here for how to do this easily in Vim: https://vim.fandom.com/wiki/Remove_unwanted_spaces

Reply:
https://github.com/MLDSAI/VisionTracing/pull/2/commits/d58445a475d4f8711cf4afb38eeb7a822cac7585